llarp/messages/common.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <llarp/util/buffer.hpp>
+
+#include <oxenc/bt.h>
+
+namespace llarp
+{
+  /// abstract base class for serialized messages
+  struct AbstractSerializable
```

Comment:
i'd love to have a bunch of free standing overloaded functions instead of using a pure virtual base class here. 

```

void
bt_encode(llarp_buffer&, const OurKindOfMessage & msg);
void
bt_encode(llarp_buffer&, const OurOtherMessage & msg);
```